front/src/applications/operationalStudies/components/MacroEditor/utils.ts

```diff
@@ -244,29 +248,98 @@ export const getTrainrunFrequencyFromTimetableItem = (
   return trainrunFrequency;
 };
 
-export const getNetzgrafikColors = () =>
-  Object.entries(NETZGRAFIK_COLOR_PALETTE).map(([colorRef, baseColor], index) => ({
-    id: index,
-    colorRef,
-    color: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
-    colorFocus: CATEGORY_COLOR_VARIANTS[`${baseColor}70`],
-    colorMuted: CATEGORY_COLOR_VARIANTS[`${baseColor}10`],
-    colorRelated: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
-    colorDarkMode: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
-    colorDarkModeFocus: CATEGORY_COLOR_VARIANTS[`${baseColor}70`],
-    colorDarkModeMuted: CATEGORY_COLOR_VARIANTS[`${baseColor}10`],
-    colorDarkModeRelated: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
-  }));
+export const getNetzgrafikColors = async ({
+  categories,
+  dispatch,
+}: {
+  categories: TrainrunCategory[];
+  dispatch: AppDispatch;
+}) => {
+  const mainColors = Object.entries(NETZGRAFIK_COLOR_PALETTE).map(
+    ([colorRef, baseColor], index) => ({
+      id: index,
+      colorRef,
+      color: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
+      colorFocus: CATEGORY_COLOR_VARIANTS[`${baseColor}70`],
+      colorMuted: CATEGORY_COLOR_VARIANTS[`${baseColor}10`],
+      colorRelated: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
+      colorDarkMode: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
+      colorDarkModeFocus: CATEGORY_COLOR_VARIANTS[`${baseColor}70`],
+      colorDarkModeMuted: CATEGORY_COLOR_VARIANTS[`${baseColor}10`],
+      colorDarkModeRelated: CATEGORY_COLOR_VARIANTS[`${baseColor}50`],
+    })
+  );
+
+  const subCategoryPage = await dispatch(
+    osrdEditoastApi.endpoints.getSubCategory.initiate({ page: 1 }, { subscribe: false })
+  ).unwrap();
 
-export const getTrainMainCategoryFromId = (id: number): TrainMainCategory | null => {
-  for (const key of OSRD_TRAINRUN_CATEGORY_MAPPING.keys()) {
-    if (OSRD_TRAINRUN_CATEGORY_MAPPING.get(key)!.id === id) {
-      return key !== 'NO_CATEGORY' ? key : null;
+  const subCategories: SubCategory[] = [...subCategoryPage.results];
+
+  const subCategoriesFromTrainCategoryToNGE = categories.filter(
+    (_, i) => i > TRAIN_MAIN_CATEGORY_TO_NGE.length - 1
+  );
```

Comment:
```suggestion
  const subTrainrunCategories = categories.filter(
    (_, i) => i > TRAIN_MAIN_CATEGORY_TO_NGE.length - 1
  );
```

Nit: this feels a bit heavy to read; also, I would prefer to keep it clear whenever we deal with a `TrainrunCategory` (which is a NGE object) or a `Category` (which is an OSRD object) - the "sub" concept can be shared

Reply:
Done